wolfcrypt/src/sha3.c

```diff
@@ -817,10 +820,15 @@ static int wc_InitSha3(wc_Sha3* sha3, void* heap, int devId)
 #if defined(WOLFSSL_ASYNC_CRYPT) && defined(WC_ASYNC_ENABLE_SHA3)
     ret = wolfAsync_DevCtxInit(&sha3->asyncDev,
                         WOLFSSL_ASYNC_MARKER_SHA3, sha3->heap, devId);
-#else
-    (void)devId;
+#elif defined(WOLF_CRYPTO_CB)
+    sha3->devId = devId;
+    sha3->type = type;
```

Comment:
In the other cases (async, sw), the p parameter passed into SHA3_Update and SHA3_Final controls which type (256, 384, 512) of SHA3 is being used.  Consider letting the the processing functions determine the info->hash->type based on that and remove the type parameter/member here.  Comments below.